crates/runix-types/src/narinfo.rs

```diff
@@ -0,0 +1,23 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::DerivationPath;
+
+fn default_true() -> bool {
+    true
+}
+
+/// Narinfo stores information output by `nix path-info --json`
+#[derive(Serialize, Deserialize, Clone)]
+pub struct Narinfo {
+    pub path: DerivationPath,
+    // TODO remove this default once https://github.com/NixOS/nix/pull/7924 has
+    // made it's way into our version of Nix
+    #[serde(default = "default_true")]
+    pub valid: bool,
+    // TODO add other fields
+    #[serde(flatten)]
+    _other: HashMap<String, Value>,
```

Comment:
Are the map values known or do they change by host package - I don't think so.
Kinda prefer making Narinfo an enum Narinfo::valid and Narinfo::invalid

Reply:
If we want to change that, can we do that as a separate PR? This is copied straight from flox-types

Reply:
If we don't need to inspect those I guess we can